src/common/libsubprocess/local.c

```diff
@@ -396,7 +396,9 @@ static void child_watch_cb (flux_reactor_t *r, flux_watcher_t *w,
 
 static int create_process (flux_subprocess_t *p)
 {
-    if (!p->hooks.pre_exec && !flux_cmd_getcwd (p->cmd))
+    if (!(p->flags & FLUX_SUBPROCESS_FLAGS_FORK_EXEC)
+        && !p->hooks.pre_exec
+        && !flux_cmd_getcwd (p->cmd))
```

Comment:
This seems fine, but is it perhaps a bit overkill to add this flag for just one test? Or were you considering other use cases? Based on the code changed here, I'm sure you saw that you can also force fork()/exec() by setting a cwd of `.`.

Reply:
yeah, it is a bit overkill :-)  But I figured there's a decent chance we'll need to force fork/exec again in the future for one reason or another.  Maybe I'll be wrong :P

Reply:
Well let's hope not (at least not from the broker)